functions/src/Models/Event.ts

```diff
@@ -0,0 +1,49 @@
+import firebase from 'firebase-admin';
+
+export class Event {
+    rsvp = {};
+    name: string = ""; 
+    date: string = "";
+    startTime: string = "";
+    endTime: string = "";
+    location:  string = "";
+    points: string = "";
+    committee: string = "";
+    type: string = "";
+    id: string = "";
+    code: string = "";
+
+    constructor(event: Partial<Event> = {}) {
+        Object.assign(this, event);
+        this.code = this.makeCode();
+    }
+
+    /**
+     * Creates a random Code with the length inputted. The code omits characters that commonly look similar in popular fonts.
+     *
+     * @access     private
+     * @param {Number=} length  The length of the code you want created.
+     */
+
+    makeCode(length = 4) {
+        let text = "";
+        let possible = "ABCDEFGHJKMNPQRSTUVWXYZ23456789";
+
+        for (let i = 0; i < length; i++)
+            text += possible.charAt(Math.floor(Math.random() * possible.length));
+
+        return text;
+    }
+}
+
```

Comment:
These comments should explain why we do this not just what it does. For the most part people can just read code and see what it does but it's much harder to know why things are being done so the majority of your comments should be comments that answer Why and maybe give an example of how things should be done. Other thing to note is that we use JsDocs to document our code. That's done with `/** */`.  Look at how we do comments on the functions on this link [here](https://github.com/SHPEUCF/shpeucfapp/blob/master/src/ducks/User.js).  You don't need to do it exactly how it's shown there because this is an object and those are functions but you more or less see how to do it.

Reply:
Just to give clarification as to why it's good to have why comments instead of what comments:
Most people spend a lot more time [reading code than writing code](https://www.google.com/search?q=time+spent+reading+code&sxsrf=ALeKk01oeXxUI0CEQz9G2KCRV3gAJFLbGg:1598211559104&source=lnms&tbm=isch&sa=X&ved=2ahUKEwj2iYaCirLrAhVuRN8KHWdCDMQQ_AUoAXoECA8QAw&biw=2560&bih=1297#imgrc=iws0yS4gDCucyMs) during development.
This means you need to write very good understandable code and good comments. When you write good code it should be easy to understand what's going on in the code. What's not easy to understand is why we made the decisions that we did a few months after we made those decisions.

Reply:
Hello please review my previous comments.